"use strict";(self.webpackChunkgenopipe=self.webpackChunkgenopipe||[]).push([[496],{3905:(e,t,r)=>{r.d(t,{Zo:()=>c,kt:()=>h});var n=r(7294);function o(e,t,r){return t in e?Object.defineProperty(e,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):e[t]=r,e}function a(e,t){var r=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),r.push.apply(r,n)}return r}function i(e){for(var t=1;t<arguments.length;t++){var r=null!=arguments[t]?arguments[t]:{};t%2?a(Object(r),!0).forEach((function(t){o(e,t,r[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(r)):a(Object(r)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(r,t))}))}return e}function l(e,t){if(null==e)return{};var r,n,o=function(e,t){if(null==e)return{};var r,n,o={},a=Object.keys(e);for(n=0;n<a.length;n++)r=a[n],t.indexOf(r)>=0||(o[r]=e[r]);return o}(e,t);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);for(n=0;n<a.length;n++)r=a[n],t.indexOf(r)>=0||Object.prototype.propertyIsEnumerable.call(e,r)&&(o[r]=e[r])}return o}var s=n.createContext({}),p=function(e){var t=n.useContext(s),r=t;return e&&(r="function"==typeof e?e(t):i(i({},t),e)),r},c=function(e){var t=p(e.components);return n.createElement(s.Provider,{value:t},e.children)},u="mdxType",f={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},d=n.forwardRef((function(e,t){var r=e.components,o=e.mdxType,a=e.originalType,s=e.parentName,c=l(e,["components","mdxType","originalType","parentName"]),u=p(r),d=o,h=u["".concat(s,".").concat(d)]||u[d]||f[d]||a;return r?n.createElement(h,i(i({ref:t},c),{},{components:r})):n.createElement(h,i({ref:t},c))}));function h(e,t){var r=arguments,o=t&&t.mdxType;if("string"==typeof e||o){var a=r.length,i=new Array(a);i[0]=d;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l[u]="string"==typeof e?e:o,i[1]=l;for(var p=2;p<a;p++)i[p]=r[p];return n.createElement.apply(null,i)}return n.createElement.apply(null,r)}d.displayName="MDXCreateElement"},6551:(e,t,r)=>{r.r(t),r.d(t,{assets:()=>s,contentTitle:()=>i,default:()=>f,frontMatter:()=>a,metadata:()=>l,toc:()=>p});var n=r(7462),o=(r(7294),r(3905));const a={id:"output",title:"",sidebar_label:"Output Files (-o)"},i="Output Report (-o)",l={unversionedId:"StrainID/output",id:"StrainID/output",title:"",description:"If StrainID loops through a directory of n BAM files, there will be n output files written to the directory path indicated by -o with file reports named XXXXXX_strain.tab. In each output file, there is a score written for each strain VCF tested. Below are two examples of StrainID reports with descriptions of how to interpret the results.",source:"@site/docs/StrainID/output.md",sourceDirName:"StrainID",slug:"/StrainID/output",permalink:"/GenoPipe-docs/docs/StrainID/output",draft:!1,editUrl:"https://github.com/CEGRcode/GenoPipe-docs/docs/StrainID/output.md",tags:[],version:"current",frontMatter:{id:"output",title:"",sidebar_label:"Output Files (-o)"},sidebar:"strainSidebar",previous:{title:"Customizing your database",permalink:"/GenoPipe-docs/docs/StrainID/db/custom-db"},next:{title:"FAQ",permalink:"/GenoPipe-docs/docs/StrainID/faq"}},s={},p=[{value:"Example 1",id:"example-1",level:2},{value:"Example 2",id:"example-2",level:2},{value:"Example 3",id:"example-3",level:2}],c={toc:p},u="wrapper";function f(e){let{components:t,...r}=e;return(0,o.kt)(u,(0,n.Z)({},c,r,{components:t,mdxType:"MDXLayout"}),(0,o.kt)("h1",{id:"output-report--o"},"Output Report (",(0,o.kt)("inlineCode",{parentName:"h1"},"-o"),")"),(0,o.kt)("p",null,"If StrainID loops through a directory of ",(0,o.kt)("inlineCode",{parentName:"p"},"n")," BAM files, there will be ",(0,o.kt)("inlineCode",{parentName:"p"},"n")," output files written to the directory path indicated by ",(0,o.kt)("inlineCode",{parentName:"p"},"-o")," with file reports named ",(0,o.kt)("inlineCode",{parentName:"p"},"XXXXXX_strain.tab"),". In each output file, there is a score written for each strain VCF tested. Below are two examples of StrainID reports with descriptions of how to interpret the results."),(0,o.kt)("h2",{id:"example-1"},"Example 1"),(0,o.kt)("p",null,'Below is an example StrainID output file from an "MCF-7" cell line sample (',(0,o.kt)("a",{parentName:"p",href:"https://www.encodeproject.org/files/ENCFF354LGE/"},"ENCFF354LGE"),")."),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre"},"    ENCFF354LGE.bam\nA549.vcf    -3.474049648043522\nLnCap.vcf   -3.5372670186299637\nK562.vcf    -2.7537084792821593\nHCT116.vcf  -3.6683635463465363\nHELA.vcf    -4.021637878101243\nSKnSH.vcf   -2.123208861240921\nHepG2.vcf   -2.707522230324179\nMCF7.vcf    2.6946153576498024\n")),(0,o.kt)("p",null,"For this report, the MCF-7 score relative to the scores of the other strains is much higher relative to the scores of other strains. Thus, this sample is classified by StrainID to be from a MCF-7 background."),(0,o.kt)("h2",{id:"example-2"},"Example 2"),(0,o.kt)("p",null,'Below is an example StrainID output file from an "SK-N-SH" cell line sample (',(0,o.kt)("a",{parentName:"p",href:"https://www.encodeproject.org/files/ENCFF000QXV/"},"ENCFF000QXV"),")."),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre"},"    ENCFF000QXV.bam\nA549.vcf    -0.5559535480469241\nLnCap.vcf   -4.600347667405377\nK562.vcf    NaN\nHCT116.vcf  NaN\nHELA.vcf    NaN\nSKnSH.vcf   4.049767512841029\nHepG2.vcf   NaN\nMCF7.vcf    -1.5507918429311454\n")),(0,o.kt)("p",null,'In reading these reports, look for the VCF file with the largest value as the best match VCF for the data. When you see the "NaN" score associated with a VCF file (K562, HCT116, HeLA, and HepG2 in this case), then the scripts found zero alternate allele matches in the data for that VCF. In these cases, it is still possible that the dataset is sourced from one of these strains on the chance that no reads mapped to the region, particularly for low coverage datasets.'),(0,o.kt)("p",null,"For this report, due to the dramatically high value of the SK-N-SH score relative to the scores of the other strains, this sample would be confidently classified as sourced from the SK-N-SH cell line."),(0,o.kt)("h2",{id:"example-3"},"Example 3"),(0,o.kt)("p",null,"Sometimes in reports you may see an 'Inf' score (should be extremely rare). In these cases, the background of the dataset was not sufficiently sampled and no reads sampled included deviations from the reference genome. You may try re-running StrainID with a different seed."),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre"},"    XXXXX.bam\nstrain1.vcf -5.400766523407889\nstrain2.vcf Inf\n")))}f.isMDXComponent=!0}}]);