"use strict";(self.webpackChunkgenopipe=self.webpackChunkgenopipe||[]).push([[53],{1109:e=>{e.exports=JSON.parse('{"pluginId":"default","version":"current","label":"Next","banner":null,"badge":false,"noIndex":false,"className":"docs-version-current","isLast":true,"docsSidebars":{"generalSidebar":[{"type":"link","label":"Welcome","href":"/GenoPipe-docs/docs/welcome","docId":"welcome"},{"type":"link","label":"Quick Start","href":"/GenoPipe-docs/docs/quick-start","docId":"quick-start"}],"epitopeSidebar":[{"type":"link","label":"Usage","href":"/GenoPipe-docs/docs/EpitopeID/","docId":"EpitopeID/epitopeid"},{"type":"link","label":"Input Files (-i)","href":"/GenoPipe-docs/docs/EpitopeID/input","docId":"EpitopeID/input"},{"type":"category","label":"Reference Database (-d)","items":[{"type":"link","label":"Pregenerated databases","href":"/GenoPipe-docs/docs/EpitopeID/db/pregenerated","docId":"EpitopeID/db/pregenerated"},{"type":"link","label":"Customizing your database","href":"/GenoPipe-docs/docs/EpitopeID/db/custom-db","docId":"EpitopeID/db/custom-db"}],"collapsed":true,"collapsible":true,"href":"/GenoPipe-docs/docs/EpitopeID/db/structure"},{"type":"link","label":"Output Files (-o)","href":"/GenoPipe-docs/docs/EpitopeID/output","docId":"EpitopeID/output"},{"type":"link","label":"Threading (-t)","href":"/GenoPipe-docs/docs/EpitopeID/performance","docId":"EpitopeID/performance"},{"type":"link","label":"Threshold (-p)","href":"/GenoPipe-docs/docs/EpitopeID/threshold","docId":"EpitopeID/threshold"},{"type":"link","label":"FAQ","href":"/GenoPipe-docs/docs/EpitopeID/faq","docId":"EpitopeID/faq"}],"deletionSidebar":[{"type":"link","label":"Usage","href":"/GenoPipe-docs/docs/DeletionID/","docId":"DeletionID/deletionid"},{"type":"link","label":"Input Files (-i)","href":"/GenoPipe-docs/docs/DeletionID/input","docId":"DeletionID/input"},{"type":"category","label":"Reference Database (-d)","items":[{"type":"link","label":"Pregenerated databases","href":"/GenoPipe-docs/docs/DeletionID/db/pregenerated","docId":"DeletionID/db/pregenerated"},{"type":"link","label":"Customizing your database","href":"/GenoPipe-docs/docs/DeletionID/db/custom-db","docId":"DeletionID/db/custom-db"}],"collapsed":true,"collapsible":true,"href":"/GenoPipe-docs/docs/DeletionID/db/structure"},{"type":"link","label":"Output Files (-o)","href":"/GenoPipe-docs/docs/DeletionID/output","docId":"DeletionID/output"},{"type":"link","label":"FAQ","href":"/GenoPipe-docs/docs/DeletionID/faq","docId":"DeletionID/faq"}],"strainSidebar":[{"type":"link","label":"Usage","href":"/GenoPipe-docs/docs/StrainID/","docId":"StrainID/strainid"},{"type":"link","label":"Input Files (-i)","href":"/GenoPipe-docs/docs/StrainID/input","docId":"StrainID/input"},{"type":"link","label":"Genome Files (-g)","href":"/GenoPipe-docs/docs/StrainID/genome","docId":"StrainID/genome"},{"type":"category","label":"Reference Database (-v)","items":[{"type":"link","label":"Pregenerated databases","href":"/GenoPipe-docs/docs/StrainID/db/pregenerated","docId":"StrainID/db/pregenerated"},{"type":"link","label":"Customizing your database","href":"/GenoPipe-docs/docs/StrainID/db/custom-db","docId":"StrainID/db/custom-db"}],"collapsed":true,"collapsible":true,"href":"/GenoPipe-docs/docs/StrainID/db/structure"},{"type":"link","label":"Output Files (-o)","href":"/GenoPipe-docs/docs/StrainID/output","docId":"StrainID/output"},{"type":"link","label":"Set Random Seed (-s)","href":"/GenoPipe-docs/docs/StrainID/seed","docId":"StrainID/seed"},{"type":"link","label":"FAQ","href":"/GenoPipe-docs/docs/StrainID/faq","docId":"StrainID/faq"}]},"docs":{"DeletionID/db/custom-db":{"id":"DeletionID/db/custom-db","title":"","description":"Customizing your database","sidebar":"deletionSidebar"},"DeletionID/db/pregenerated":{"id":"DeletionID/db/pregenerated","title":"","description":"Pregenerated databases","sidebar":"deletionSidebar"},"DeletionID/db/structure":{"id":"DeletionID/db/structure","title":"","description":"For DeletionID, this is the \\"database\\" or directory with all the reference files used by identify-Deletion.sh. DeletionID provides reference files for yeast (sacCer3_Del) so you can quickly get started without building up the database from scratch. However, you are free to customize the database by adding a different set of coordinate new mappability scores or by looking at a different set of coordinate intervals.","sidebar":"deletionSidebar"},"DeletionID/deletionid":{"id":"DeletionID/deletionid","title":"Usage","description":"DeletionID identifies significant depletions of aligned NGS tags in the genome relative to a background model. This tool is set-up to confirm full gene knockouts.","sidebar":"deletionSidebar"},"DeletionID/faq":{"id":"DeletionID/faq","title":"FAQ","description":"Q: I have run DeletionID but many genes are being returned in the output report. Are all of these genes in the list depleted from my sample?","sidebar":"deletionSidebar"},"DeletionID/input":{"id":"DeletionID/input","title":"","description":"DeletionID takes a directory pathname for the input and will run DeletionID on all the BAM files (*.bam) it finds within that directory path when you execute identify-Deletion.sh. Make sure your files are sorted and use the standard BAM format (output of most aligner tools).","sidebar":"deletionSidebar"},"DeletionID/output":{"id":"DeletionID/output","title":"","description":"The output report is saved to the user-provided output directory in a file named based on the input BAM files (/path/to/output/XXXXX_deletion.tab from some input XXXXX.bam). Below is a sample report based on the results from running DeletionID on a whole genome sequencing dataset of the APE3 deleted knockout from the Yeast Knockout Collection (ERS838258 sample downloaded from ENA and generated by Puddu et al, 2019).","sidebar":"deletionSidebar"},"EpitopeID/db/custom-db":{"id":"EpitopeID/db/custom-db","title":"","description":"Customizing your database","sidebar":"epitopeSidebar"},"EpitopeID/db/pregenerated":{"id":"EpitopeID/db/pregenerated","title":"","description":"Pregenerated databases","sidebar":"epitopeSidebar"},"EpitopeID/db/structure":{"id":"EpitopeID/db/structure","title":"","description":"Whether you use the provided reference files or create your own, the database should use the following directory structure to ensure that EpitopeID can find the correct reference files.","sidebar":"epitopeSidebar"},"EpitopeID/epitopeid":{"id":"EpitopeID/epitopeid","title":"Usage","description":"EpitopeID identifies and determines the genomic location of epitopes or other inserted sequences relative to genomic loci.","sidebar":"epitopeSidebar"},"EpitopeID/faq":{"id":"EpitopeID/faq","title":"FAQ","description":"Q: My epitope sequence isn\'t part of the sequences in the default provided reference files (either sacCer3EpiID or hg19EpiID). Can I still use EpitopeID for checking my samples?","sidebar":"epitopeSidebar"},"EpitopeID/input":{"id":"EpitopeID/input","title":"","description":"EpitopeID takes gzipped FASTQ files from single-end(SE) or paired-end(PE) datasets to run. If your FASTQ files are not already compressed, you can zip them yourself if gzip is installed:","sidebar":"epitopeSidebar"},"EpitopeID/output":{"id":"EpitopeID/output","title":"","description":"The output report is saved to the user-provided output directory in a file named based on the input FASTQ files (/path/to/output/XXXXX_R1-ID.tab).","sidebar":"epitopeSidebar"},"EpitopeID/performance":{"id":"EpitopeID/performance","title":"","description":"This optional input is used to specify the number of threads to used for the Bowtie2 alignment commands in the identify-Epitope.sh script. Defaults to 1.","sidebar":"epitopeSidebar"},"EpitopeID/threshold":{"id":"EpitopeID/threshold","title":"","description":"This optional input is used to specify the threshold for a localization hit to be included in the report. Defaults to 0.05.","sidebar":"epitopeSidebar"},"quick-start":{"id":"quick-start","title":"Quick Start","description":"These are quick setup instructions for running each of the GenoPipe modules \\"out-of-the-box\\" using the provided pre-generated sacCer3 and hg19 reference files. For more details on customizations or reading reports, check out the module-specific pages (EpitopeID, DeletionID, and StrainID)","sidebar":"generalSidebar"},"StrainID/db/custom-db":{"id":"StrainID/db/custom-db","title":"","description":"Customizing your VCFs","sidebar":"strainSidebar"},"StrainID/db/pregenerated":{"id":"StrainID/db/pregenerated","title":"","description":"Pregenerated VCF sets","sidebar":"strainSidebar"},"StrainID/db/structure":{"id":"StrainID/db/structure","title":"","description":"For StrainID, this is the \\"database\\" or directory with all the VCFs used by identify-Strain.sh. You will notice that StrainID provides reference files for both yeast and human so you can quickly get started without building up the database from scratch. However, you are free to customize the database by adding your own VCF files.","sidebar":"strainSidebar"},"StrainID/faq":{"id":"StrainID/faq","title":"FAQ","description":"Q: What do the \\"NaN\\" or \\"Inf\\" scores mean in the StrainID output report?","sidebar":"strainSidebar"},"StrainID/genome":{"id":"StrainID/genome","title":"","description":"A reference genome is required for StrainID to determine the background model of mutation rates. Include the filepath for to the genomic FASTA file for the parameter value. Make sure the genome used matches the genome the VCF coordinates are based off of.","sidebar":"strainSidebar"},"StrainID/input":{"id":"StrainID/input","title":"","description":"StrainID takes a directory pathname for the input and will run StrainID on all the BAM files (*.bam) it finds within that directory path when you execute identify-Strain.sh.","sidebar":"strainSidebar"},"StrainID/output":{"id":"StrainID/output","title":"","description":"If StrainID loops through a directory of n BAM files, there will be n output files written to the directory path indicated by -o with file reports named XXXXXX_strain.tab. In each output file, there is a score written for each strain VCF tested. Below are two examples of StrainID reports with descriptions of how to interpret the results.","sidebar":"strainSidebar"},"StrainID/seed":{"id":"StrainID/seed","title":"","description":"This option allows the user to set a random seed for reproducible random results.","sidebar":"strainSidebar"},"StrainID/strainid":{"id":"StrainID/strainid","title":"Usage","description":"StrainID compares a database of VCF files against an aligned BAM file to check for the presence of SNPs in order to determine a likely cell line/strain used in the experiment.","sidebar":"strainSidebar"},"vignette":{"id":"vignette","title":"vignette","description":"The ENCODE experiment ENCSR692RET is a ChIP-seq experiment on a K562 cell line sample that has been CRISPR modified with an eGFP insertion to target the NR4A1 gene protein product. This vignette will show how EpitopeID and StrainID can be run on the same sample."},"welcome":{"id":"welcome","title":"Welcome","description":"Welcome to the GenoPipe wiki!","sidebar":"generalSidebar"}}}')}}]);