<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-strainid">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<title data-rh="true">StrainID | GenoPipe</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://CEGRcode.github.io/GenoPipe/docs/strainid"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="StrainID | GenoPipe"><meta data-rh="true" name="description" content="StrainID compares a database of VCF files against an aligned BAM file to check for the presence of SNPs in order to determine a likely cell line/strain used in the experiment."><meta data-rh="true" property="og:description" content="StrainID compares a database of VCF files against an aligned BAM file to check for the presence of SNPs in order to determine a likely cell line/strain used in the experiment."><link data-rh="true" rel="icon" href="/GenoPipe/img/genopipe.ico"><link data-rh="true" rel="canonical" href="https://CEGRcode.github.io/GenoPipe/docs/strainid"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/GenoPipe/docs/strainid" hreflang="en"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/GenoPipe/docs/strainid" hreflang="x-default"><link rel="stylesheet" href="/GenoPipe/assets/css/styles.f96a4498.css">
<link rel="preload" href="/GenoPipe/assets/js/runtime~main.38ea104a.js" as="script">
<link rel="preload" href="/GenoPipe/assets/js/main.33f988fd.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/GenoPipe/"><div class="navbar__logo"><img src="/GenoPipe/img/genopipe.png" alt="GenoPipe Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/GenoPipe/img/genopipe.png" alt="GenoPipe Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">GenoPipe</b></a><a class="navbar__item navbar__link" href="/GenoPipe/docs/welcome">General</a><a class="navbar__item navbar__link" href="/GenoPipe/docs/epitopeid">EpitopeID</a><a class="navbar__item navbar__link" href="/GenoPipe/docs/deletionid">DeletionID</a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/GenoPipe/docs/strainid">StrainID</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode/GenoPipe" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><main class="docMainContainer_uL0j docMainContainerEnhanced_oQfM"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>StrainID</h1></header><p><strong>StrainID compares a database of VCF files against an aligned BAM file to check for the presence of SNPs in order to determine a likely cell line/strain used in the experiment.</strong></p><p>A report is generated that scores each input BAM with a value for each given VCF file such thataslkdjfal;skjdf alsdkjf</p><p><img loading="lazy" alt="Figure1C" src="/GenoPipe/assets/images/figure1c-3571e76b934ab5d28d797b11a774f758.png" width="1893" height="458" class="img_E7b_"></p><h2 class="anchor anchorWithStickyNavbar_mojV" id="usage">Usage<a class="hash-link" href="#usage" title="Direct link to heading">​</a></h2><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">identify-Strain.sh -i /path/to/BAM -g /path/to/genome/fASTA -v /path/to/VCF/files -o /path/to/output</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="input-i">Input(<code>-i</code>)<a class="hash-link" href="#input-i" title="Direct link to heading">​</a></h2><p>StrainID takes a directory pathname for the input and will run StrainID on all the BAM files (<code>*.bam</code>) it finds within that directory path when you execute <code>identify-Strain.sh</code>.</p><div class="admonition admonition-caution alert alert--warning"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</h5></div><div class="admonition-content"><p>It is extremely important that the BAM files provided match the reference genome build that the VCF files are based off of.</p></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="reference-genome--g">Reference Genome (<code>-g</code>)<a class="hash-link" href="#reference-genome--g" title="Direct link to heading">​</a></h2><p>A reference genome is required for StrainID to determine the background model of mutation rates. Include the filepath for to the genomic FASTA file for the parameter value. Make sure the genome used matches the genome the VCF coordinates are based off of.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="reference-files--v">Reference Files (<code>-v</code>)<a class="hash-link" href="#reference-files--v" title="Direct link to heading">​</a></h2><p>For StrainID, this is the &quot;database&quot; or directory with all the VCFs used by <code>identify-Strain.sh</code>. You will notice that StrainID provides reference files for both yeast (<code>sacCer3_VCF</code>) and human (<code>hg19_VCF</code>) so you can quickly get started without building up the database from scratch. However, you are free to customize the database by adding your own VCF files.</p><div class="admonition admonition-caution alert alert--warning"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</h5></div><div class="admonition-content"><p>If you add your own VCF files, make sure they match the genome build of both your BAM files and the other VCF files.</p></div></div><p>Below describes the yeast and human VCF databases and lists which strain files come with the StrainID tool.</p><ul><li>For yeast, the VCF-DB (<code>sacCer3_VCF/*.vcf</code>) with strains listed below</li><li>For humans, the VCF-DB (<code>hg19_VCF/*.vcf</code>) with strains listed below</li></ul><table><thead><tr><th align="center"><code>sacCer3_VCF</code></th><th align="center"><code>hg19_VCF</code></th></tr></thead><tbody><tr><td align="center">BY4741</td><td align="center">A549</td></tr><tr><td align="center">BY4742</td><td align="center">HCT116</td></tr><tr><td align="center">CEN.PK2-1Ca</td><td align="center">HELA</td></tr><tr><td align="center">D273-10B</td><td align="center">HepG2</td></tr><tr><td align="center">JK9-3d</td><td align="center">K562</td></tr><tr><td align="center">RedStar</td><td align="center">LnCap</td></tr><tr><td align="center">RM11-1A</td><td align="center">MCF7</td></tr><tr><td align="center">SEY6210</td><td align="center">SKnSH</td></tr><tr><td align="center">Sigma1278b-10560-6B</td><td align="center"></td></tr><tr><td align="center">SK1</td><td align="center"></td></tr><tr><td align="center">W303</td><td align="center"></td></tr><tr><td align="center">Y55</td><td align="center"></td></tr></tbody></table><p>Whether you use the provided reference files or create your own, the database should use the following directory structure both to ensure that StrainID can find the correct reference files and for simplicity, clarity, and consistency.</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">/name/of/strain/vcf/db</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--strain1.vcf</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--strain2.vcf</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--strain3.vcf</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--strain4.vcf</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="customizing-your-vcfs">Customizing your VCFs<a class="hash-link" href="#customizing-your-vcfs" title="Direct link to heading">​</a></h3><p>Adding strains is very straightforward. Simply add a file in standard <a href="https://www.google.com" target="_blank" rel="noopener noreferrer">Variant Call Format</a> to the <code>/name/of/strain/vcf/db/</code> directory.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="output-report--o">Output Report (<code>-o</code>)<a class="hash-link" href="#output-report--o" title="Direct link to heading">​</a></h2><p>If StrainID loops through a directory of <code>n</code> BAM files, there will be <code>n</code> output files written to the directory path indicated by <code>-o</code> with file reports named <code>XXXXXX_strain.tab</code>.</p><p>Below is an example output file where there is a score written to each strain VCF provided with a significant score.</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">  ENCFF000QXV.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">README.txt  NaN</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">LnCap.vcf   -4.523561956057013</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">MCF7.vcf    -3.321928094887362</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">SKnSH.vcf   2.415037499278844</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">A549.vcf    -3.0</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="admonition admonition-warning alert alert--danger"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>warning</h5></div><div class="admonition-content"><p>fill in output report description</p><p>how some scores have lower precision---relatedness between VCFs and</p></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="faqs">FAQs<a class="hash-link" href="#faqs" title="Direct link to heading">​</a></h2><ul><li>Q: I used the 14 VCF files in the default sacCer_VCF database but the output only prints out score for 7 of the strains.
A: First, check the error messages to make sure StrainID isn&#x27;t quitting part way for any reason (such as hitting resource allocation limits). If StrainID has completed the run with no issues, this can still be the expected behavior. StrainID does not print scores in circumstances such as if there are no reads for alternate allele signal (no)</li></ul></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/CEGRcode/GenoPipe-docs/docs/strainid.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#usage" class="table-of-contents__link toc-highlight">Usage</a></li><li><a href="#input-i" class="table-of-contents__link toc-highlight">Input(<code>-i</code>)</a></li><li><a href="#reference-genome--g" class="table-of-contents__link toc-highlight">Reference Genome (<code>-g</code>)</a></li><li><a href="#reference-files--v" class="table-of-contents__link toc-highlight">Reference Files (<code>-v</code>)</a><ul><li><a href="#customizing-your-vcfs" class="table-of-contents__link toc-highlight">Customizing your VCFs</a></li></ul></li><li><a href="#output-report--o" class="table-of-contents__link toc-highlight">Output Report (<code>-o</code>)</a></li><li><a href="#faqs" class="table-of-contents__link toc-highlight">FAQs</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Module Docs</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/GenoPipe/docs/epitopeid">EpitopeID</a></li><li class="footer__item"><a class="footer__link-item" href="/GenoPipe/docs/deletionid">DeletionID</a></li><li class="footer__item"><a class="footer__link-item" href="/GenoPipe/docs/strainid">StrainID</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu/" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab Website<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/CEGRcode/scriptmanager" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/CEGRcode/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/CEGRcode/GenoPipe" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 GenoPipe. Built with Docusaurus.</div></div></div></footer></div>
<script src="/GenoPipe/assets/js/runtime~main.38ea104a.js"></script>
<script src="/GenoPipe/assets/js/main.33f988fd.js"></script>
</body>
</html>