<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-strainid">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v2.3.1">
<title data-rh="true">StrainID | GenoPipe</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://CEGRcode.github.io/GenoPipe-docs/docs/strainid"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="StrainID | GenoPipe"><meta data-rh="true" name="description" content="StrainID compares a database of VCF files against an aligned BAM file to check for the presence of SNPs in order to determine a likely cell line/strain used in the experiment."><meta data-rh="true" property="og:description" content="StrainID compares a database of VCF files against an aligned BAM file to check for the presence of SNPs in order to determine a likely cell line/strain used in the experiment."><link data-rh="true" rel="icon" href="/GenoPipe-docs/img/genopipe.ico"><link data-rh="true" rel="canonical" href="https://CEGRcode.github.io/GenoPipe-docs/docs/strainid"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/GenoPipe-docs/docs/strainid" hreflang="en"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/GenoPipe-docs/docs/strainid" hreflang="x-default"><link rel="stylesheet" href="/GenoPipe-docs/assets/css/styles.945767d3.css">
<link rel="preload" href="/GenoPipe-docs/assets/js/runtime~main.84a52057.js" as="script">
<link rel="preload" href="/GenoPipe-docs/assets/js/main.5a30fd46.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region" aria-label="Skip to main content"><a class="skipToContent_fXgn" href="#docusaurus_skipToContent_fallback">Skip to main content</a></div><nav aria-label="Main" class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Toggle navigation bar" aria-expanded="false" class="navbar__toggle clean-btn" type="button"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/GenoPipe-docs/"><div class="navbar__logo"><img src="/GenoPipe-docs/img/genopipe.png" alt="GenoPipe Logo" class="themedImage_ToTc themedImage--light_HNdA"><img src="/GenoPipe-docs/img/genopipe.png" alt="GenoPipe Logo" class="themedImage_ToTc themedImage--dark_i4oU"></div><b class="navbar__title text--truncate">GenoPipe</b></a><a class="navbar__item navbar__link" href="/GenoPipe-docs/docs/welcome">General</a><a class="navbar__item navbar__link" href="/GenoPipe-docs/docs/epitopeid">EpitopeID</a><a class="navbar__item navbar__link" href="/GenoPipe-docs/docs/deletionid">DeletionID</a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/GenoPipe-docs/docs/strainid">StrainID</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode/GenoPipe" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_vylO colorModeToggle_DEke"><button class="clean-btn toggleButton_gllP toggleButtonDisabled_aARS" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)" aria-live="polite"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_pyhR"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_wfgR"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_ZlJk"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div id="docusaurus_skipToContent_fallback" class="main-wrapper mainWrapper_z2l0 docsWrapper_BCFX"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_sjWU" type="button"></button><div class="docPage__5DB"><main class="docMainContainer_gTbr docMainContainerEnhanced_Uz_u"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_VOVn"><div class="docItemContainer_Djhp"><article><div class="tocCollapsible_ETCw theme-doc-toc-mobile tocMobile_ITEo"><button type="button" class="clean-btn tocCollapsibleButton_TO0P">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>StrainID</h1></header><p><strong>StrainID compares a database of VCF files against an aligned BAM file to check for the presence of SNPs in order to determine a likely cell line/strain used in the experiment.</strong></p><p>A report is generated that scores each input BAM with a value for each given VCF file such thataslkdjfal;skjdf alsdkjf</p><p><img loading="lazy" alt="Figure1C" src="/GenoPipe-docs/assets/images/figure1c-3571e76b934ab5d28d797b11a774f758.png" width="1893" height="458" class="img_ev3q"></p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="usage">Usage<a href="#usage" class="hash-link" aria-label="Direct link to Usage" title="Direct link to Usage">​</a></h2><div class="language-bash codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-bash codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">identify-Strain.sh -i /path/to/BAM -g /path/to/genome/fASTA -v /path/to/VCF/files -o /path/to/output</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="input-i">Input(<code>-i</code>)<a href="#input-i" class="hash-link" aria-label="Direct link to input-i" title="Direct link to input-i">​</a></h2><p>StrainID takes a directory pathname for the input and will run StrainID on all the BAM files (<code>*.bam</code>) it finds within that directory path when you execute <code>identify-Strain.sh</code>.</p><div class="theme-admonition theme-admonition-caution alert alert--warning admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</div><div class="admonitionContent_S0QG"><p>It is extremely important that the BAM files provided match the reference genome build that the VCF files are based off of.</p></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="reference-genome--g">Reference Genome (<code>-g</code>)<a href="#reference-genome--g" class="hash-link" aria-label="Direct link to reference-genome--g" title="Direct link to reference-genome--g">​</a></h2><p>A reference genome is required for StrainID to determine the background model of mutation rates. Include the filepath for to the genomic FASTA file for the parameter value. Make sure the genome used matches the genome the VCF coordinates are based off of.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="reference-files--v">Reference Files (<code>-v</code>)<a href="#reference-files--v" class="hash-link" aria-label="Direct link to reference-files--v" title="Direct link to reference-files--v">​</a></h2><p>For StrainID, this is the &quot;database&quot; or directory with all the VCFs used by <code>identify-Strain.sh</code>. You will notice that StrainID provides reference files for both yeast (<code>sacCer3_VCF</code>) and human (<code>hg19_VCF</code>) so you can quickly get started without building up the database from scratch. However, you are free to customize the database by adding your own VCF files.</p><div class="theme-admonition theme-admonition-caution alert alert--warning admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</div><div class="admonitionContent_S0QG"><p>If you add your own VCF files, make sure they match the genome build of both your BAM files and the other VCF files.</p></div></div><p>The table below lists the strain VCFs provided yeast and human databases that come with the StrainID tool.</p><ul><li>For yeast, the VCF-DB (<code>sacCer3_VCF/*.vcf</code>) with strains listed below</li><li>For humans, the VCF-DB (<code>hg19_VCF/*.vcf</code>) with strains listed below</li></ul><table><thead><tr><th align="center"><code>sacCer3_VCF</code></th><th align="center"><code>hg19_VCF</code></th></tr></thead><tbody><tr><td align="center">BY4741</td><td align="center">A549</td></tr><tr><td align="center">BY4742</td><td align="center">HCT116</td></tr><tr><td align="center">CEN.PK2-1Ca</td><td align="center">HELA</td></tr><tr><td align="center">D273-10B</td><td align="center">HepG2</td></tr><tr><td align="center">JK9-3d</td><td align="center">K562</td></tr><tr><td align="center">RedStar</td><td align="center">LnCap</td></tr><tr><td align="center">RM11-1A</td><td align="center">MCF7</td></tr><tr><td align="center">SEY6210</td><td align="center">SKnSH</td></tr><tr><td align="center">Sigma1278b-10560-6B</td><td align="center"></td></tr><tr><td align="center">SK1</td><td align="center"></td></tr><tr><td align="center">W303</td><td align="center"></td></tr><tr><td align="center">Y55</td><td align="center"></td></tr></tbody></table><h3 class="anchor anchorWithStickyNavbar_LWe7" id="customizing-your-vcfs">Customizing your VCFs<a href="#customizing-your-vcfs" class="hash-link" aria-label="Direct link to Customizing your VCFs" title="Direct link to Customizing your VCFs">​</a></h3><p>Adding strains is very straightforward. Simply add a file in standard <a href="https://genome.ucsc.edu/goldenPath/help/vcf.html" target="_blank" rel="noopener noreferrer">Variant Call Format</a> to the <code>/name/of/strain/vcf/db/</code> directory. The database should use the following directory structure both to ensure that StrainID can find the correct reference files.</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">/name/of/strain/vcf/db</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--strain1.vcf</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--strain2.vcf</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--strain3.vcf</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--strain4.vcf</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="output-report--o">Output Report (<code>-o</code>)<a href="#output-report--o" class="hash-link" aria-label="Direct link to output-report--o" title="Direct link to output-report--o">​</a></h2><p>If StrainID loops through a directory of <code>n</code> BAM files, there will be <code>n</code> output files written to the directory path indicated by <code>-o</code> with file reports named <code>XXXXXX_strain.tab</code>. In each output file, there is a score written for each strain VCF tested. Below are two examples of StrainID reports with descriptions of how to interpret the results.</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="example-1">Example 1<a href="#example-1" class="hash-link" aria-label="Direct link to Example 1" title="Direct link to Example 1">​</a></h3><p>Below is an example StrainID output file from an &quot;MCF-7&quot; cell line sample (<a href="https://www.encodeproject.org/files/ENCFF354LGE/" target="_blank" rel="noopener noreferrer">ENCFF354LGE</a>).</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">    ENCFF354LGE.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">A549.vcf    -3.474049648043522</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">LnCap.vcf   -3.5372670186299637</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">K562.vcf    -2.7537084792821593</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">HCT116.vcf  -3.6683635463465363</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">HELA.vcf    -4.021637878101243</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">SKnSH.vcf   -2.123208861240921</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">HepG2.vcf   -2.707522230324179</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">MCF7.vcf    2.6946153576498024</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>For this report, the MCF-7 score relative to the scores of the other strains is much higher relative to the scores of other strains. Thus, this sample is classified by StrainID to be from a MCF-7 background.</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="example-2">Example 2<a href="#example-2" class="hash-link" aria-label="Direct link to Example 2" title="Direct link to Example 2">​</a></h3><p>Below is an example StrainID output file from an &quot;SK-N-SH&quot; cell line sample (<a href="https://www.encodeproject.org/files/ENCFF000QXV/" target="_blank" rel="noopener noreferrer">ENCFF000QXV</a>).</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">    ENCFF000QXV.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">A549.vcf    -0.5559535480469241</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">LnCap.vcf   -4.600347667405377</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">K562.vcf    NaN</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">HCT116.vcf  NaN</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">HELA.vcf    NaN</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">SKnSH.vcf   4.049767512841029</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">HepG2.vcf   NaN</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">MCF7.vcf    -1.5507918429311454</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>In reading these reports, look for the VCF file with the largest value as the best match VCF for the data. When you see the &quot;NaN&quot; score associated with a VCF file (K562, HCT116, HeLA, and HepG2 in this case), then the scripts found zero alternate allele matches in the data for that VCF. In these cases, it is still possible that the dataset is sourced from one of these strains on the chance that no reads mapped to the region, particularly for low coverage datasets.</p><p>For this report, due to the dramatically high value of the SK-N-SH score relative to the scores of the other strains, this sample would be confidently classified as sourced from the SK-N-SH cell line.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="faqs">FAQs<a href="#faqs" class="hash-link" aria-label="Direct link to FAQs" title="Direct link to FAQs">​</a></h2><ul><li>Q: What do the &quot;NaN&quot; or &quot;Inf&quot; scores mean in the StrainID output report?<ul><li>Please check the examples with descriptions in the section describing the <a href="/GenoPipe-docs/docs/strainid#output-report--o">output reports</a>.</li></ul></li><li>Q: All of my strains are reporting &quot;NaN&quot;. Does my sample truly not match any of the strains I gave it?<ul><li>Make sure your VCF files are non-empty</li><li>Genome build match: make sure your BAM files are aligned to the same genome build your VCF files are based on. If you performed a LiftOver to get your VCF file, there might be some unresolved issues with the chain file that caused some misalignments.</li><li>Make sure your read coverage is decent across the VCF alleles</li><li>It&#x27;s still possible to return a &quot;NaN&quot; for all VCF files if no data mapped to the alleles you are checking for</li></ul></li><li>Q: I customized a set of VCF files to use with StrainID but it is taking forever to run. Is there a way to speed things up?<ul><li>You may be using large VCF files across a large genome. The larger the VCF, the longer StrainID will take to run. Consider subsampling the VCF or filtering to only include high-confidence or unique alleles. You do not need a complete set of all alleles to calculate this QC heuristic.</li></ul></li></ul></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/CEGRcode/GenoPipe-docs/docs/strainid.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_Z9Sw" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_vwxv"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"></nav></div></div><div class="col col--3"><div class="tableOfContents_bqdL thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#usage" class="table-of-contents__link toc-highlight">Usage</a></li><li><a href="#input-i" class="table-of-contents__link toc-highlight">Input(<code>-i</code>)</a></li><li><a href="#reference-genome--g" class="table-of-contents__link toc-highlight">Reference Genome (<code>-g</code>)</a></li><li><a href="#reference-files--v" class="table-of-contents__link toc-highlight">Reference Files (<code>-v</code>)</a><ul><li><a href="#customizing-your-vcfs" class="table-of-contents__link toc-highlight">Customizing your VCFs</a></li></ul></li><li><a href="#output-report--o" class="table-of-contents__link toc-highlight">Output Report (<code>-o</code>)</a><ul><li><a href="#example-1" class="table-of-contents__link toc-highlight">Example 1</a></li><li><a href="#example-2" class="table-of-contents__link toc-highlight">Example 2</a></li></ul></li><li><a href="#faqs" class="table-of-contents__link toc-highlight">FAQs</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Modules</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/GenoPipe-docs/docs/epitopeid">EpitopeID</a></li><li class="footer__item"><a class="footer__link-item" href="/GenoPipe-docs/docs/deletionid">DeletionID</a></li><li class="footer__item"><a class="footer__link-item" href="/GenoPipe-docs/docs/strainid">StrainID</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/GenoPipe-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="margin-bottom--sm"><img src="/GenoPipe-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--light_HNdA footer__logo"><img src="/GenoPipe-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--dark_i4oU footer__logo"></div><div class="footer__copyright">Copyright © 2023 GenoPipe. Built with Docusaurus.</div></div></div></footer></div>
<script src="/GenoPipe-docs/assets/js/runtime~main.84a52057.js"></script>
<script src="/GenoPipe-docs/assets/js/main.5a30fd46.js"></script>
</body>
</html>