<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-epitopeid">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<title data-rh="true">EpitopeID | GenoPipe</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://github.com/CEGRcode/GenoPipe/docs/epitopeid"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="EpitopeID | GenoPipe"><meta data-rh="true" name="description" content="EpitopeID identifies and determines the genomic location of epitopes or other inserted sequences relative to genomic loci."><meta data-rh="true" property="og:description" content="EpitopeID identifies and determines the genomic location of epitopes or other inserted sequences relative to genomic loci."><link data-rh="true" rel="icon" href="/GenoPipe/img/genopipe.ico"><link data-rh="true" rel="canonical" href="https://github.com/CEGRcode/GenoPipe/docs/epitopeid"><link data-rh="true" rel="alternate" href="https://github.com/CEGRcode/GenoPipe/docs/epitopeid" hreflang="en"><link data-rh="true" rel="alternate" href="https://github.com/CEGRcode/GenoPipe/docs/epitopeid" hreflang="x-default"><link rel="stylesheet" href="/GenoPipe/assets/css/styles.f96a4498.css">
<link rel="preload" href="/GenoPipe/assets/js/runtime~main.38ea104a.js" as="script">
<link rel="preload" href="/GenoPipe/assets/js/main.d284abf2.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/GenoPipe/"><div class="navbar__logo"><img src="/GenoPipe/img/genopipe.png" alt="GenoPipe Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/GenoPipe/img/genopipe.png" alt="GenoPipe Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">GenoPipe</b></a><a class="navbar__item navbar__link" href="/GenoPipe/docs/welcome">General</a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/GenoPipe/docs/epitopeid">EpitopeID</a><a class="navbar__item navbar__link" href="/GenoPipe/docs/deletionid">DeletionID</a><a class="navbar__item navbar__link" href="/GenoPipe/docs/strainid">StrainID</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode/GenoPipe" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><main class="docMainContainer_uL0j docMainContainerEnhanced_oQfM"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>EpitopeID</h1></header><p><strong>EpitopeID identifies and determines the genomic location of epitopes or other inserted sequences relative to genomic loci.</strong></p><p><img loading="lazy" alt="Figure1A" src="/GenoPipe/assets/images/figure1a-f28316ada6378d6d3299c663bfe961eb.png" width="1894" height="459" class="img_E7b_"></p><h2 class="anchor anchorWithStickyNavbar_mojV" id="usage">Usage<a class="hash-link" href="#usage" title="Direct link to heading">​</a></h2><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> identify-Epitope.sh -i /path/to/FASTQ -o /path/to/output -d /path/to/genome/database -t </span><span class="token operator" style="color:#393A34">&lt;</span><span class="token plain">Threads - Default </span><span class="token operator file-descriptor important" style="color:#393A34">1</span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">eg: </span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> identify-Epitope.sh -i /input -o /output -d /sacCer3_EpiID -t </span><span class="token number" style="color:#36acaa">2</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="specific-dependencies">Specific Dependencies<a class="hash-link" href="#specific-dependencies" title="Direct link to heading">​</a></h2><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">conda create -n epitopeid -c bioconda bowtie2 bedtools samtools pysam scipy</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="input--i">Input (<code>-i</code>)<a class="hash-link" href="#input--i" title="Direct link to heading">​</a></h2><p>EpitopeID takes <a href="https://www.gnu.org/software/gzip/manual/gzip.html" target="_blank" rel="noopener noreferrer">gzipped</a> FASTQ files from single-end(SE) or paired-end(PE) datasets to run. If your FASTQ files are not already compressed, you can zip them yourself if gzip is installed:</p><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token function" style="color:#d73a49">gzip</span><span class="token plain"> XXXXX_R1.fastq</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>It is expected that at least one file for each sample (even for single-end data) follows the naming convention of <code>XXXXX_R1*.fastq.gz</code> and a second file if the data is paired-end following <code>XXXXX_R2*.fastq.gz</code>. This is based on the Illumina naming standard.</p><div class="admonition admonition-caution alert alert--warning"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</h5></div><div class="admonition-content"><p>Make sure none of the sample names used in the filenames have an occurrence of <code>_R1</code> outside of the read-specifying term to avoid confusion for EpitopeID when it is determining which samples have a read2 file.</p><p>For example, samples named like this will cause confusion:<br>
❌ <code>Sample_R13A_R1.fastq.gz</code></p><p>Alternatively try:<br>
✅ <code>Sample-R13A_R1.fastq.gz</code><br>
✅ <code>SampleR13A_R1.fastq.gz</code></p></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="reference-files--d">Reference Files (<code>-d</code>)<a class="hash-link" href="#reference-files--d" title="Direct link to heading">​</a></h2><p>For EpitopeID, this is the &quot;database&quot; or directory with four types of reference files used by <code>identify-Epitope.sh</code>. You will notice that EpitopeID provides reference files for both yeast (<code>sacCer3_EpiID</code>) and human (<code>hg19_EpiID</code>) so you can quickly get started without building up the database from scratch. However, you are free to customize and build your own set of files (e.g. add different epitope tags to check for, use a different genome build).</p><div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</h5></div><div class="admonition-content"><p>The provided database files are missing the genomic reference file for storage reasons. You will need to follow the directions below to download <code>genome.fa</code> before running EpitopeID if you are planning to use the provided defaults.</p></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="database-structure">Database structure<a class="hash-link" href="#database-structure" title="Direct link to heading">​</a></h3><p>Whether you use the provided reference files or create your own, the database should use the following directory structure for the sake of organization, clarity, and consistency, and ensuring that EpitopeID can find the correct reference files.</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">/name/of/epiDB</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--FASTA_tag</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--Tag_DB</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |  |--tagname1.fa</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |  |--tagname2.fa</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |  |--tagname3.fa</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--ALL_TAG.fa</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--ALL_TAG.fa.1.bt2       # EpitopeID will automatically generate these</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--ALL_TAG.fa.2.bt2       # Bowtie2 index files if any are missing</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--ALL_TAG.fa.3.bt2       #</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--ALL_TAG.fa.4.bt2       #</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--ALL_TAG.fa.rev.1.bt2   #</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--ALL_TAG.fa.rev.2.bt2   #</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--FASTA_genome</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--genome.fa              # Not included, download separately</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--annotation</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--genome_annotation.gff.gz</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--blacklist_filter</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|  |--blacklist.bed</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Below is a list of the the hardcoded filenames that EpitopeID looks for during execution and some information on the provided yeast and human defaults.</p><ul><li>The <code>FASTA_tag/ALL_TAG.fa</code> is the FASTA formatted collection of all epitope sequences to search for. The <a href="https://github.com/CEGRcode/GenoPipe/tree/master/EpitopeID/sacCer3_EpiID/FASTA_tag/Tag_DB" target="_blank" rel="noopener noreferrer">yeast tag database</a> includes the AID, Extended-Tap,  FRB, HA_v1, HA_v3, MNase, ProtA, CBP, FLAG-3x, GFP, HA_v2, HaloTag, and Myc(3x) sequences. The <a href="https://www.google.com" target="_blank" rel="noopener noreferrer">human tag database</a> only includes the LAP(eGFP) tag but it is easy to customize the list to include other epitopes for EpitopeID to look for.</li><li>The <code>FASTA_genome/genome.fa</code> is the reference sequence used for the genomic alignments and the sequence that the below genome annotations are based on.</li><li>The <code>annotation/genome_annotation.gff.gz</code> file defines the bin coordinates to use when localizing the epitope insertion in PE datasets. The yeast default uses SGD gene annotation coordinates to defines one bin for the length of each gene, 250bp bins flanking each set of gene coordinates, and 250bp bins breaking up the remaining intergenic regions. The human default similarly bins out the genome using 1000bp windows on the NCBI Refseq annotations.</li><li>The <code>blacklist_filter/blacklist.bed</code> allows the user to ignore artefactual or unwanted alignments.</li></ul><p>Below is more information on how to use the utility scripts to download and customize your reference files.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="downloading-the-saccer3-genome">Downloading the sacCer3 genome<a class="hash-link" href="#downloading-the-saccer3-genome" title="Direct link to heading">​</a></h3><p>Use the utility scripts by following the commands below to download the sacCer3 genome and format the chromosome names. The reference files are based on the arabic numeral chromosome naming system (i.e. &quot;chr1&quot;, &quot;chr2&quot;,...&quot;chr16&quot;,&quot;chrM&quot;,&quot;2-micron&quot;).</p><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token builtin class-name">cd</span><span class="token plain"> /path/to/GenoPipe/EpitopeID/utility_scripts/genome_data</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> download_sacCer3_Genome.sh</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">mv</span><span class="token plain"> genome.fa /path/to/sacCer3_EpiID/FASTA_genome/</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="downloading-the-hg19-genome">Downloading the hg19 genome<a class="hash-link" href="#downloading-the-hg19-genome" title="Direct link to heading">​</a></h3><p>Use the utility scripts by following the commands below to download the hg19 genome and move it to the appropriate directory.</p><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token builtin class-name">cd</span><span class="token plain"> /path/to/GenoPipe/EpitopeID/utility_scripts/genome_data</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> download_hg19_Genome.sh</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">mv</span><span class="token plain"> genome.fa /path/to/hg19_EpiID/FASTA_genome/genome.fa</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="customizing-epitopes">Customizing epitopes<a class="hash-link" href="#customizing-epitopes" title="Direct link to heading">​</a></h3><p>If you choose to add or remove epitope tags to your database, you must add/remove the files with the sequences in FASTA format to/from <code>FASTA_tag/TagDB</code> and recreate the <code>FASTA_tag/ALL_TAG.fa</code> file so it includes the tag sequences you want. The following commands show you how to copy <code>my-custom-tag.fa</code> into the database and remake the file using the provided scripts.</p><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token comment" style="color:#999988;font-style:italic"># Copy or remove the FASTA files with your tag sequences into the Tag_DB directory</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">cp</span><span class="token plain"> my-custom-tag.fa /path/to/Tag_DB</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token builtin class-name">cd</span><span class="token plain"> /name/of/epiDB/FASTA_tag/Tag_DB</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> /path/to/GenoPipe/EpitopeID/utility_scripts/update_TagDB.sh</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">mv</span><span class="token plain"> ALL_TAG.fa* /path/to/hg19_EpiID/FASTA_tag/</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="customizing-annotations">Customizing annotations<a class="hash-link" href="#customizing-annotations" title="Direct link to heading">​</a></h3><p>GenoPipe provides the utility scripts for recreating the precomputed reference annotation files. The scripts download (yeast and human) annotations and for format gene annotation files to the EpitopeID format by tiling the genome around and including gene intervals.</p><p>The precomputed files should work for most (yeast and human) use cases but if you need to compute these reference files yourself, use the available <code>utility_scripts</code> as follows:</p><h4 class="anchor anchorWithStickyNavbar_mojV" id="make-genome_annotationgffgz-with-a-different-bin-size">Make <code>genome_annotation.gff.gz</code> with a different bin size<a class="hash-link" href="#make-genome_annotationgffgz-with-a-different-bin-size" title="Direct link to heading">​</a></h4><p>If you wish to change the bin size of the tiles from the reference files GenoPipe already provides, you can rerun our scripts with a different value in the <code>-b</code> flag. The following are the specific commands you can execute to do so.</p><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token comment" style="color:#999988;font-style:italic"># sacCer3</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token builtin class-name">cd</span><span class="token plain"> /path/to/GenoPipe/EpitopeID/utility_scripts/annotation_data</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> generate_sacCer3_GenomeAnnotation.sh -g /path/to/genome/sacCer3.fa -b </span><span class="token operator" style="color:#393A34">&lt;</span><span class="token plain">BIN_SIZE</span><span class="token operator" style="color:#393A34">&gt;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token comment" style="color:#999988;font-style:italic"># hg19</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token builtin class-name">cd</span><span class="token plain"> /path/to/GenoPipe/EpitopeID/utility_scripts/annotation_data</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> generate_hg19_GenomeAnnotation.sh -g /path/to/genome/hg19.fa -b </span><span class="token operator" style="color:#393A34">&lt;</span><span class="token plain">BIN_SIZE</span><span class="token operator" style="color:#393A34">&gt;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token comment" style="color:#999988;font-style:italic"># hg38</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token builtin class-name">cd</span><span class="token plain"> /path/to/GenoPipe/EpitopeID/utility_scripts/annotation_data</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> generate_hg38_GenomeAnnotation.sh -g /path/to/genome/hg38.fa -b </span><span class="token operator" style="color:#393A34">&lt;</span><span class="token plain">BIN_SIZE</span><span class="token operator" style="color:#393A34">&gt;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h4 class="anchor anchorWithStickyNavbar_mojV" id="make-genome_annotationgffgz-with-custom-annotations">Make <code>genome_annotation.gff.gz</code> with custom annotations<a class="hash-link" href="#make-genome_annotationgffgz-with-custom-annotations" title="Direct link to heading">​</a></h4><p>There may be a few reasons to create a custom annotation reference set:</p><ul><li>Working with non-yeast and non-human data</li><li>Inserted sequence localized to non-ORF genomic location (e.g. insertions in enhancer region)</li><li>Inserted sequence/epitope localized to ORF not included in precomputed annotations (rare, for genes not included in official set at the time we created the precomputed files)</li></ul><div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</h5></div><div class="admonition-content"><p>EpitopeID actually can still detect and localize insertions from non-ORF regions but the report will only include the nearest ORF or genomic tile and may be more difficult to read/interpret. Creating a customized annotation reference file would improve clarity in the output report but is not <em>necessary</em>.</p></div></div><ol><li>Create a custom <code>.gff</code> file including the feature with the expected localization.</li></ol><ul><li>It may be a good idea to include other potential off-target annotations for better readability of the report. Otherwise off-target localizations will be reported with only the genomic coordinate information.</li><li>If you are working with a custom genome build, you will need the gene annotations in <code>.gff</code> format for the genome build. <a href="https://useast.ensembl.org/info/data/ftp/index.html" target="_blank" rel="noopener noreferrer">Ensembl</a> and <a href="https://hgdownload.soe.ucsc.edu/downloads.html" target="_blank" rel="noopener noreferrer">UCSC</a> can be good resources for finding gene annotations associated with official genome builds across a variety of organisms. Please make sure they are in <code>.gff</code> file format for compatibility with the utility scripts.</li></ul><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain"># Example .gff entry like an enhancer or something</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># myunicorn_annotations.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><ol start="2"><li>Execute the following commands to tile the genome and merge the annotations with the tiled regions.</li></ol><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token assign-left variable" style="color:#36acaa">MYGFF</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/myunicorn_annotations.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">GENOME</span><span class="token operator" style="color:#393A34">=</span><span class="token plain">/path/to/genome/unicorn1.fa</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Choose a bin size (consider size of genome)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token assign-left variable" style="color:#36acaa">BIN</span><span class="token operator" style="color:#393A34">=</span><span class="token number" style="color:#36acaa">250</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Add flanking sequence to</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">perl parsers/parse_Generic_GFF.pl </span><span class="token variable" style="color:#36acaa">$MYGFF</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$BIN</span><span class="token plain"> temp.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Tile genome</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">perl genome_bin/bin_genome.pl </span><span class="token variable" style="color:#36acaa">$GENOME</span><span class="token plain"> </span><span class="token variable" style="color:#36acaa">$BIN</span><span class="token plain"> unicorn1_BIN.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Intersect gene annotations</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">bedtools subtract -a unicorn1_BIN.gff -b temp.gff </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> unicorn1_BIN_temp.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Merge annotations and bin file</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">perl genome_bin/rename_BIN_GFF.pl unicorn1_BIN_temp.gff unicorn1_BIN_filter.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">cat</span><span class="token plain"> unicorn1_BIN_filter.gff temp.gff </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> unicorn1_ALL.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">sort</span><span class="token plain"> -k </span><span class="token number" style="color:#36acaa">1,1</span><span class="token plain"> -k4,4n unicorn1_ALL.gff </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> genome_annotation.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Compress</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">gzip</span><span class="token plain"> -f genome_annotation.gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Clean-up</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">rm</span><span class="token plain"> temp.gff unicorn1_BIN.gff unicorn1_BIN_temp.gff unicorn1_BIN_filter.gff unicorn1_ALL.gff</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="customizing-filter">Customizing filter<a class="hash-link" href="#customizing-filter" title="Direct link to heading">​</a></h3><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token comment" style="color:#999988;font-style:italic"># bedtools intersect -v -abam $OUTPUT/$SAMPLE/orf.bam -b $DATABASE/blacklist_filter/blacklist.bed &gt; $OUTPUT/$SAMPLE/orf_filter.bam</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">mv</span><span class="token plain"> /path/to/blacklist_filter.bed /path/to/EpiID-DB/blacklist_filter/blacklist.bed</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="output-report--o">Output Report (<code>-o</code>)<a class="hash-link" href="#output-report--o" title="Direct link to heading">​</a></h2><p>The output report is saved to the user-provided output directory in a file named based on the input FASTQ files (<code>/path/to/output/XXXXX_R1-ID.tab</code>). Below is a sample report based on the results from running EpitopeID on the ENCODE <a href="https://www.encodeproject.org/files/ENCFF415CJF/" target="_blank" rel="noopener noreferrer">ENCFF415CJF</a> sample.</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">EpitopeID   EpitopeCount</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">LAP-tag 435</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">GeneID  EpitopeID   EpitopeLocation EpitopeCount    pVal</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">NR4A1|chr12:52416616-52453291   LAP-tag C-term  9   3.580493355965414e-24</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The first part of the report shows which epitopes in <code>Tag_DB</code> were identified in the sample (<strong>EpitopeID column</strong>) and how many reads mapped to this epitope (<strong>EpitopeCount</strong>) to help quantify the coverage of the epitopes which relates to the confidence of the call.</p><p>The second part of the report shows which epitopes localized to which regions/tiles of the genome significantly (sorted by pvalue if multiple hits). The columns specify the coordinate interval (<strong>GeneID</strong>), which epitope maps to this locus (<strong>EpitopeID</strong>), if this occurs on the N or C-terminus (<strong>EpitopeLocation</strong>), the number of reads mapping to this tile (<strong>EpitopeCount</strong>), and the poisson-calculated associated p-value to indicate confidence of the site (<strong>pVal</strong>).</p><p>In this report there were 435 reads that mapped to the LAP-tag epitope. The high read count supports &quot;the epitope was successfully integrated into the sample.&quot;&quot;</p><p>In this report there were 9 read pairs that spanned both the LAP-tag and the NR4A1 gene(at genomic locus chr12:52416616-52453291) toward the sequence encoding the C-terminus. This entry supports that the LAP-tag epitope was integrated at the C-terminus of NF4A1.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="threading--t">Threading (<code>-t</code>)<a class="hash-link" href="#threading--t" title="Direct link to heading">​</a></h2><p>This optional input is used to specify the number of threads to used for the Bowtie2 alignment commands. Defaults to 1.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="example-set-up-epitopeid-and-run-on-yeast-example">Example: Set-up EpitopeID and run on yeast example<a class="hash-link" href="#example-set-up-epitopeid-and-run-on-yeast-example" title="Direct link to heading">​</a></h2><div class="language-bash codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-bash codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token function" style="color:#d73a49">git</span><span class="token plain"> clone www.github/CEGRcode/GenoPipe</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token builtin class-name">cd</span><span class="token plain"> GenoPipe/EpitopeID/</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># remake the epitope Tags file and index it (if custom sequences are added)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token builtin class-name">cd</span><span class="token plain"> sacCer3_EpiID/FASTA_tag/Tag_DB</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><span class="token comment" style="color:#999988;font-style:italic">## add custom sequences here ##</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/</span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/</span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/utility_scripts/update_TagDB.sh</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">mv</span><span class="token plain"> ALL_TAG.fa* </span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># download the genome</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token builtin class-name">cd</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/</span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/</span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/utility_scripts/genome_data/</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> download_sacCer3_Genome.sh</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">mv</span><span class="token plain"> genome.fa </span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/</span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/sacCer3_EpiID/FASTA_genome/</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># download annotations</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># cd ../annotation_data/</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># bash generate_sacCer3_GenomeAnnotation.sh -g ../../sacCer3_EpiID/FASTA_genome/genome.fa -b 250</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># mv genome_annotation.gff.gz ../../sacCer3_EpiID/annotation/</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token builtin class-name">cd</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/</span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># run EpitopeID on FASTQs in the `sample` directory below</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">mkdir</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/output</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> identify_Epitope.sh -i </span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/samples/ -o </span><span class="token punctuation" style="color:#393A34">..</span><span class="token plain">/output/ -d sacCer3_EpiID -t </span><span class="token number" style="color:#36acaa">4</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="faqs">FAQs<a class="hash-link" href="#faqs" title="Direct link to heading">​</a></h2><ul><li>Q: My epitope sequence isn&#x27;t part of the sequences in the default provided reference files (either <code>sacCer3_EpiID</code> or <code>hg19_EpiID</code>). Can I still use EpitopeID for checking my samples?<ul><li>Yes! Please scroll up to the <a href="/GenoPipe/docs/epitopeid#customizing-epitopes"><strong>Customizing epitopes</strong></a> section for directions on how to add your epitope sequence to the database.</li></ul></li><li>Q: I added my own custom tag sequences to the <code>TagDB</code> directory but when I run EpitopeID, none of my samples are getting significant hits to the new tags.<ul><li>There are a few things you should check before concluding that the epitope is not present in your sample:<ol><li>Did you recreate the <code>ALL_TAG.fa</code> file? Open it up to make sure your sequences are there. If they aren&#x27;t there, follow the commands in the &quot;How to add your own epitope tag sequences&quot; section above.</li><li>Did you recreate the Bowtie2 index files for <code>ALL_TAG.fa</code>? Remove at least one of the Bowtie2 index files (e.g. <code>ALL_TAG.1.bt2</code>) and run EpitopeID. EpitopeID automatically recreates the index if any files are missing but it does not check that the index files match the FASTA sequence. If you modify the <code>ALL_TAG.fa</code> without recreating the BWA index files, EpitopeID will run on the old set of tag sequences.</li></ol></li></ul></li><li>Q: What does the N-term or C-term mean in the output?<ul><li>For each gene, three bins are created for (1) the ORF interval, (2) a bin upstream of the start codon, (3) a bin downstream of the stop codon, each of which correspond to the gene, N-terminus, or C-terminus of the peptide chain gene product. If EpitopeID is mapping more strongly to the N or C-terminus, it is likely that the epitope is tagged to that side of the endogenous peptide chain.</li></ul></li><li>Q: I tried running EpitopeID on my samples using <code>sacCer3_EpiID</code>/<code>hg19_EpiID</code> and it is telling me that I am missing the <code>genomic.fa</code> file. Where can I find it?<ul><li>Due to storage reasons, we do not include the genomic reference in the Github download for GenoPipe. However, we do provide scripts that will help you to download and format the reference genomes that we used. See above for directions on how to use them by looking at the section titled &quot;Downloading the sacCer3/hg19 genome&quot;. Otherwise you can use your own genomic sequence. If you do, then remember to rename it to <code>genome.fa</code> and move it to the appropriate directory. Also make sure that the annotations you use are based on the same reference build.</li></ul></li></ul></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/CEGRcode/GenoPipe-docs/docs/epitopeid.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#usage" class="table-of-contents__link toc-highlight">Usage</a></li><li><a href="#specific-dependencies" class="table-of-contents__link toc-highlight">Specific Dependencies</a></li><li><a href="#input--i" class="table-of-contents__link toc-highlight">Input (<code>-i</code>)</a></li><li><a href="#reference-files--d" class="table-of-contents__link toc-highlight">Reference Files (<code>-d</code>)</a><ul><li><a href="#database-structure" class="table-of-contents__link toc-highlight">Database structure</a></li><li><a href="#downloading-the-saccer3-genome" class="table-of-contents__link toc-highlight">Downloading the sacCer3 genome</a></li><li><a href="#downloading-the-hg19-genome" class="table-of-contents__link toc-highlight">Downloading the hg19 genome</a></li><li><a href="#customizing-epitopes" class="table-of-contents__link toc-highlight">Customizing epitopes</a></li><li><a href="#customizing-annotations" class="table-of-contents__link toc-highlight">Customizing annotations</a></li><li><a href="#customizing-filter" class="table-of-contents__link toc-highlight">Customizing filter</a></li></ul></li><li><a href="#output-report--o" class="table-of-contents__link toc-highlight">Output Report (<code>-o</code>)</a></li><li><a href="#threading--t" class="table-of-contents__link toc-highlight">Threading (<code>-t</code>)</a></li><li><a href="#example-set-up-epitopeid-and-run-on-yeast-example" class="table-of-contents__link toc-highlight">Example: Set-up EpitopeID and run on yeast example</a></li><li><a href="#faqs" class="table-of-contents__link toc-highlight">FAQs</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Module Docs</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/GenoPipe/docs/epitopeid">EpitopeID</a></li><li class="footer__item"><a class="footer__link-item" href="/GenoPipe/docs/deletionid">DeletionID</a></li><li class="footer__item"><a class="footer__link-item" href="/GenoPipe/docs/strainid">StrainID</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu/" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab Website<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/CEGRcode/scriptmanager" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/CEGRcode/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/CEGRcode/GenoPipe" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 GenoPipe. Built with Docusaurus.</div></div></div></footer></div>
<script src="/GenoPipe/assets/js/runtime~main.38ea104a.js"></script>
<script src="/GenoPipe/assets/js/main.d284abf2.js"></script>
</body>
</html>