<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-deletionid">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v2.3.1">
<title data-rh="true">DeletionID | GenoPipe</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://CEGRcode.github.io/GenoPipe-docs/docs/deletionid"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="DeletionID | GenoPipe"><meta data-rh="true" name="description" content="DeletionID identifies significant depletions of aligned NGS tags in the genome relative to a background model. This tool is set-up to confirm full gene knockouts."><meta data-rh="true" property="og:description" content="DeletionID identifies significant depletions of aligned NGS tags in the genome relative to a background model. This tool is set-up to confirm full gene knockouts."><link data-rh="true" rel="icon" href="/GenoPipe-docs/img/genopipe.ico"><link data-rh="true" rel="canonical" href="https://CEGRcode.github.io/GenoPipe-docs/docs/deletionid"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/GenoPipe-docs/docs/deletionid" hreflang="en"><link data-rh="true" rel="alternate" href="https://CEGRcode.github.io/GenoPipe-docs/docs/deletionid" hreflang="x-default"><link rel="stylesheet" href="/GenoPipe-docs/assets/css/styles.945767d3.css">
<link rel="preload" href="/GenoPipe-docs/assets/js/runtime~main.84a52057.js" as="script">
<link rel="preload" href="/GenoPipe-docs/assets/js/main.5a30fd46.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region" aria-label="Skip to main content"><a class="skipToContent_fXgn" href="#docusaurus_skipToContent_fallback">Skip to main content</a></div><nav aria-label="Main" class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Toggle navigation bar" aria-expanded="false" class="navbar__toggle clean-btn" type="button"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/GenoPipe-docs/"><div class="navbar__logo"><img src="/GenoPipe-docs/img/genopipe.png" alt="GenoPipe Logo" class="themedImage_ToTc themedImage--light_HNdA"><img src="/GenoPipe-docs/img/genopipe.png" alt="GenoPipe Logo" class="themedImage_ToTc themedImage--dark_i4oU"></div><b class="navbar__title text--truncate">GenoPipe</b></a><a class="navbar__item navbar__link" href="/GenoPipe-docs/docs/welcome">General</a><a class="navbar__item navbar__link" href="/GenoPipe-docs/docs/epitopeid">EpitopeID</a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/GenoPipe-docs/docs/deletionid">DeletionID</a><a class="navbar__item navbar__link" href="/GenoPipe-docs/docs/strainid">StrainID</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/CEGRcode/GenoPipe" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link"></a><div class="toggle_vylO colorModeToggle_DEke"><button class="clean-btn toggleButton_gllP toggleButtonDisabled_aARS" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)" aria-live="polite"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_pyhR"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_wfgR"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_ZlJk"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div id="docusaurus_skipToContent_fallback" class="main-wrapper mainWrapper_z2l0 docsWrapper_BCFX"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_sjWU" type="button"></button><div class="docPage__5DB"><main class="docMainContainer_gTbr docMainContainerEnhanced_Uz_u"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_VOVn"><div class="docItemContainer_Djhp"><article><div class="tocCollapsible_ETCw theme-doc-toc-mobile tocMobile_ITEo"><button type="button" class="clean-btn tocCollapsibleButton_TO0P">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>DeletionID</h1></header><p><strong>DeletionID identifies significant depletions of aligned NGS tags in the genome relative to a background model. This tool is set-up to confirm full gene knockouts.</strong></p><p><img loading="lazy" alt="Figure1B" src="/GenoPipe-docs/assets/images/figure1b-398c1c0556d113c7a588ae5aae8b7762.png" width="1893" height="456" class="img_ev3q"></p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="usage">Usage<a href="#usage" class="hash-link" aria-label="Direct link to Usage" title="Direct link to Usage">​</a></h2><div class="language-bash codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-bash codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> identify-Deletion.sh -i /path/to/BAM -o /path/to/output -d /path/to/genome/database</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">eg: </span><span class="token function" style="color:#d73a49">bash</span><span class="token plain"> identify-Deletion.sh -i /input -o /output -d /sacCer3_Del</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="input-i">Input(<code>-i</code>)<a href="#input-i" class="hash-link" aria-label="Direct link to input-i" title="Direct link to input-i">​</a></h2><p>DeletionID takes a directory pathname for the input and will run DeletionID on all the BAM files (<code>*.bam</code>) it finds within that directory path when you execute <code>identify-Deletion.sh</code>. Make sure your files are sorted and use the standard BAM format (output of most aligner tools).</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="supported-assays">Supported assays<a href="#supported-assays" class="hash-link" aria-label="Direct link to Supported assays" title="Direct link to Supported assays">​</a></h3><p>Like the other GenoPipe modules, DeletionID supports input from a variety of assays. Chromatin-based assays like <strong>whole genome sequencing (WGS)</strong> and <strong>ChIP-based assays</strong> (e.g. ChIP-seq and ChIP-exo) for which we have validated this tool are appropriate inputs. The background signal of ChIP-based assays  and the smoothing effect of the binning strategy that DeletionID implements makes detection of deletions possible. Generally, genomic sequencing assays work well and transcriptomic assays are to be avoided for this module.</p><div class="theme-admonition theme-admonition-caution alert alert--warning admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</div><div class="admonitionContent_S0QG"><p>Using <strong>RNA-seq</strong> data is not recommended!</p><p>Because DeletionID works by looking for significant depletion of reads in the genome, NGS assays with read distributions across the genome that dramatically deviate from uniform coverage will not work well. The background calculation assumes similar read coverage across all intervals tested. If you are trying to validate deletion backgrounds for RNA-seq data, we suggest you use a tool like DESeq2 which compares expression scores from your deletion RNA-seq sample against a WT RNA-seq control to flag depletions.</p></div></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="supported-genomes">Supported genomes<a href="#supported-genomes" class="hash-link" aria-label="Direct link to Supported genomes" title="Direct link to Supported genomes">​</a></h3><div class="theme-admonition theme-admonition-info alert alert--info admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 14 16"><path fill-rule="evenodd" d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg></span>info</div><div class="admonitionContent_S0QG"><p>If you are trying to run DeletionID on samples from <strong>human</strong> or another organism with a larger genome than yeast, please read this!</p></div></div><p>DeletionID has been tested and validated in <strong>yeast (sacCer3)</strong> for &gt;3M paired-end datasets but has not been thoroughly tested in organisms with larger genomes. Some considerations for using DeletionID on non-yeast organisms:</p><ul><li><strong>Larger genomes</strong> will require a higher amount of sequencing to cover the genome and establish a consistent background model for checking relative depletion. Many intervals may be reported as significant that are not truly knocked out if there is insufficient coverage</li><li>The <strong>size of the gene &quot;knockout&quot; mutation</strong> relative to the size of the genome may result in poor identification of true deletions. Practically speaking, null mutants for larger genomes are typically not full gene knockouts but rather a mutation with a dramatic deleterious effect which result in a nonfunctional protein (e.g. frameshift, early termination codon, etc).</li><li><strong>Repetitive regions</strong> can cause mappability problems, creating falsely called depleted regions or influencing the background thresholding to hide significant depletions. Regions with low mappability may need to be blacklisted in the analysis.</li></ul><p>Please keep this in mind when running DeletionID and try to use high coverage datasets to confirm knockouts.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="reference-files--d">Reference Files (<code>-d</code>)<a href="#reference-files--d" class="hash-link" aria-label="Direct link to reference-files--d" title="Direct link to reference-files--d">​</a></h2><p>For DeletionID, this is the &quot;database&quot; or directory with all the reference files used by <code>identify-Deletion.sh</code>. DeletionID provides reference files for yeast (<code>sacCer3_Del</code>) so you can quickly get started without building up the database from scratch. However, you are free to customize the database by adding a different set of coordinate new mappability scores or by looking at a different set of coordinate intervals.</p><p>Below is a list of the files that DeletionID looks for during execution and some information on the provided yeast and human defaults.</p><ul><li>The <code>coord.bed</code> is the file with the genomic coordinate intervals we are looking for depletions within. The provided <code>sacCer3_Del</code> uses the set of yeast gene coordinate intervals.</li><li>The <code>mappability.tab</code> is the file with the calculated mappability scores for each coordinate interval within the <code>coord.bed</code> file. This will need to be regenerated if you use a different <code>coord.bed</code> file from the one provided.</li></ul><p>Whether you use the provided reference files or create your own, the database should use the following directory structure both to ensure that DeletionID can find the correct reference files.</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">/name/of/delDB</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--genomic_coord</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   |--coord.bed</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--mappability</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   |--mappability.tab</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Below is more information on how to use the utility scripts to download and customize your reference files.</p><h3 class="anchor anchorWithStickyNavbar_LWe7" id="how-to-add-custom-coordinate-bed-file">How to add custom coordinate BED file<a href="#how-to-add-custom-coordinate-bed-file" class="hash-link" aria-label="Direct link to How to add custom coordinate BED file" title="Direct link to How to add custom coordinate BED file">​</a></h3><p>For custom coordinates, add your BED-formatted genomic intervals for your expected deletions to <code>/name/of/delDB/genomic_coord/coord.bed</code> file. In order to make sure DeletionID calculates an appropriate background score, the <code>coord.bed</code> file should also contain a set of other &quot;background&quot; intervals of similar features or random coordinates. A handful of coordinates is not sufficient to calculate a robust background calculation so make sure you sample a good number of intervals to create the reference file.</p><p>For example, the provided <code>sacCer3_Del</code> database contains the ORF intervals for most protein-coding features in yeast. For this reference, the median coverage across all ORFs in <code>coords.bed</code> determine the background threshold for depletion calls.</p><p>Next you must generate new mappability scores for the new intervals. (See how to generate new mappabilities for a set of coordinates below)</p><div class="theme-admonition theme-admonition-caution alert alert--warning admonition_LlT9"><div class="admonitionHeading_tbUL"><span class="admonitionIcon_kALy"><svg viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</div><div class="admonitionContent_S0QG"><p>Identifying deletions of highly <strong>repetitive regions</strong> of the genome using DeletionID is not recommended!</p><p>Highly repetitive regions of the genome have very low mappability scores and are often thrown out by DeletionID even before calculating the depletion score. If you wish to proceed, please try to expand the interval (<code>coords.bed</code>) to include the entire repeat region and lower the reporting threshold in the Python script so that it reports all scores. You may want to compare the score of the region to the scores of a control sample. Your results may be strengthened using replicates, perhaps across several studies to determine if the numbers can be used to measure dramatic expansions. This has not been validated or benchmarked and should be used with caution.</p></div></div><h3 class="anchor anchorWithStickyNavbar_LWe7" id="how-to-generate-a-new-mappability-reference-file">How to generate a new mappability reference file<a href="#how-to-generate-a-new-mappability-reference-file" class="hash-link" aria-label="Direct link to How to generate a new mappability reference file" title="Direct link to How to generate a new mappability reference file">​</a></h3><p>For each set of coordinates, a companion mappability file must be generated for DeletionID. The following code shows how to generate this reference file using the coordinate interval file and the genome FASTA that the coordinates are based on.</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">cd /path/to/GenoPipe/DeletionID/utility_scripts</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">bash generate_Mappability_File.sh -f /path/to/genome.fa -c /path/to/delDB/genomic_coord/coord.bed -t &lt;Threads&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">mv mappability.tab /path/to/delDB/mappability/mappability.tab</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_LWe7" id="output-report--o">Output Report (<code>-o</code>)<a href="#output-report--o" class="hash-link" aria-label="Direct link to output-report--o" title="Direct link to output-report--o">​</a></h2><p>The output report is saved to the user-provided output directory in a file named based on the input BAM files (<code>/path/to/output/XXXXX_deletion.tab</code> from some input <code>XXXXX.bam</code>). Below is a sample report based on the results from running DeletionID on a whole genome sequencing dataset of the APE3 deleted knockout from the Yeast Knockout Collection (ERS838258 sample downloaded from ENA and generated by Puddu et al, 2019).</p><div class="codeBlockContainer_Ckt0 theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_biex"><pre tabindex="0" class="prism-code language-text codeBlock_bY9V thin-scrollbar"><code class="codeBlockLines_e6Vv"><span class="token-line" style="color:#393A34"><span class="token plain">LEU2 No Data Detected</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">URA3 No Data Detected</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">APE3 -3.3992739779627357</span><br></span></code></pre><div class="buttonGroup__atx"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_eSgA" aria-hidden="true"><svg class="copyButtonIcon_y97N" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_LjdS" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The first column of the report lists out the intervals (ORFs) with significantly depleted reads, sorted by the DeletionID score values in the second column which sort the more negative scores (more read depleted intervals) first. This sample is from an APE3 knockout in a LEU2 and URA3 knockout background so we expect all three gene intervals to show up in the report. LEU2 and URA3 show <strong>No Data Detected</strong> which indicates no reads mapped to this interval. APE3 shows a very strong depletion with a large negative score (-3.3992...) confirming the knockout as likely.</p><h2 class="anchor anchorWithStickyNavbar_LWe7" id="faqs">FAQs<a href="#faqs" class="hash-link" aria-label="Direct link to FAQs" title="Direct link to FAQs">​</a></h2><ul><li>Q: I have run DeletionID but many genes are being returned in the output report. Are all of these genes in the list depleted from my sample?<ul><li>First check your sequencing coverage. When it is low, DeletionID can misidentify or identify many more knockout regions than are actually present.</li><li>Is your assay chromatin based? DeletionID does not work well with RNAseq datasets due to the heavily skewed distribution of reads across the genome. Please read the <a href="/GenoPipe-docs/docs/deletionid#supported-assays">disclaimer</a> above for more information.</li></ul></li><li>Q: I have data from a human/mouse deletion that I wish to identify. Do you have a reference database for genes from the mm10 or hg19 genome assemblies?<ul><li>DeletionID has not been benchmarked against samples from human or other organisms with large genomes. Please see the <a href="/GenoPipe-docs/docs/deletionid#supported-genomes">caution note</a> above for more information.</li></ul></li></ul></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/CEGRcode/GenoPipe-docs/docs/deletionid.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_Z9Sw" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_vwxv"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"></nav></div></div><div class="col col--3"><div class="tableOfContents_bqdL thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#usage" class="table-of-contents__link toc-highlight">Usage</a></li><li><a href="#input-i" class="table-of-contents__link toc-highlight">Input(<code>-i</code>)</a><ul><li><a href="#supported-assays" class="table-of-contents__link toc-highlight">Supported assays</a></li><li><a href="#supported-genomes" class="table-of-contents__link toc-highlight">Supported genomes</a></li></ul></li><li><a href="#reference-files--d" class="table-of-contents__link toc-highlight">Reference Files (<code>-d</code>)</a><ul><li><a href="#how-to-add-custom-coordinate-bed-file" class="table-of-contents__link toc-highlight">How to add custom coordinate BED file</a></li><li><a href="#how-to-generate-a-new-mappability-reference-file" class="table-of-contents__link toc-highlight">How to generate a new mappability reference file</a></li></ul></li><li><a href="#output-report--o" class="table-of-contents__link toc-highlight">Output Report (<code>-o</code>)</a></li><li><a href="#faqs" class="table-of-contents__link toc-highlight">FAQs</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Modules</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/GenoPipe-docs/docs/epitopeid">EpitopeID</a></li><li class="footer__item"><a class="footer__link-item" href="/GenoPipe-docs/docs/deletionid">DeletionID</a></li><li class="footer__item"><a class="footer__link-item" href="/GenoPipe-docs/docs/strainid">StrainID</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://pughlab.mbg.cornell.edu" target="_blank" rel="noopener noreferrer" class="footer__link-item">Pugh Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://williamkmlai.github.io" target="_blank" rel="noopener noreferrer" class="footer__link-item">Lai Lab<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div><div class="col footer__col"><div class="footer__title">Other Tools We Develop</div><ul class="footer__items clean-list"><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/scriptmanager-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">ScriptManager<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="https://github.com/seqcode/pegr" target="_blank" rel="noopener noreferrer" class="footer__link-item">PEGR<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/stencil" target="_blank" rel="noopener noreferrer" class="footer__link-item">STENCIL<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li><li class="footer__item"><a href="http://pughlab.mbg.cornell.edu/GenoPipe-docs" target="_blank" rel="noopener noreferrer" class="footer__link-item">GenoPipe<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="margin-bottom--sm"><img src="/GenoPipe-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--light_HNdA footer__logo"><img src="/GenoPipe-docs/img/pughlab_logo.svg" alt="Pugh Lab Logo" class="themedImage_ToTc themedImage--dark_i4oU footer__logo"></div><div class="footer__copyright">Copyright © 2023 GenoPipe. Built with Docusaurus.</div></div></div></footer></div>
<script src="/GenoPipe-docs/assets/js/runtime~main.84a52057.js"></script>
<script src="/GenoPipe-docs/assets/js/main.5a30fd46.js"></script>
</body>
</html>